### DISCUSSION TITLE
DreamBerd 3 Formatting Style

### DISCUSSION THREAD

**Discussion by GoldenStack** [TodePond/GulfOfMexico#419] [Q&A] [answered]
hi
some situations involving parsing code are made very difficult or, i believe, impossible, if code formatting is not verified.

additionally if formatting is not standardized it makes AI (specifically AQMI, AEMI, etc) very difficult to implement as there are likely multiple possible interpretations of the code, but formatting standardization would make it, i believe, possible to implement.

would you say it's fair for a parser to make sure all code follows standard practices, like indentation?
this would mean the following code does not compile:
```java
if true { return 7! }
```
this also does not compile:
```java
if true {
   print Hi !
} print Hi 2 !
```

additionally it would render the curly brackets unnecessary for parsing while also keeping them in the langauge, which i think would be funny
(please do not remove curly brackets. i think it could make parsing slightly easier but it would cause great suffering to all of society)

**Comment by TodePond** [comment, marked as answer]
> impossible

It's all possible! :) It's supposed to be difficult. Sorry, gonna keep it as it is

**Reply by GoldenStack** [reply]
oh i didn't mean to be asking for it to be changed! thanks for answering anyway :D
